akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala

```diff
@@ -462,33 +462,41 @@ class ActorLoggingSpec extends ScalaTestWithActorTestKit("""
     }
 
     "always include some MDC values in the log" in {
-      // need AtomicReference because LoggingFilter defined before actor is created and ActorTestKit names are dynamic
-      val actorPathStr = new AtomicReference[String]
       val behavior =
         Behaviors.setup[Message] { context =>
-          actorPathStr.set(context.self.path.toString)
           context.log.info("Starting")
           Behaviors.receiveMessage { _ =>
-            if (MDC.get("logSource") != null)
-              throw new IllegalStateException("MDC wasn't cleared. logSource has value before context.log is accessed.")
+            for (key <- ("logSource" :: "akkaSource" :: "tags" :: Nil)) {
+              if (MDC.get(key) != null)
```

Comment:
Or perhaps just
```scala
MDC.get("logSource") shouldEqual null
```

Reply:
We'd want to include what key was not cleared so that's why it is like this